src/main/java/gov/cms/madie/hqmf/HQMFGeneratorFactory.java

```diff
@@ -0,0 +1,21 @@
+package gov.cms.madie.hqmf;
+
+import gov.cms.madie.hqmf.qdm_5_6.HQMFGenerator;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+/** Factory class to select the proper HQMF generator based on QDM model version. */
+@Slf4j
+@Component
+@AllArgsConstructor
+public class HQMFGeneratorFactory {
+
+  private HQMFGenerator qdm_5_6_generator;
+
+  public Generator getHQMFGenerator() {
+    // Keeping generator around in case we need to support additional versions or implementations in
+    // the future
```

Comment:
Comment is not relevant in this context. It mattered in the MAT where we needed to be able generate HQMF files for older versions of QDM.

Reply:
hope we don't get another version of QDM(5.7)

Reply:
Honestly, if we do, it might better to replace this flow with something more manageable.